test/new-e2e/tests/ndm/snmp/snmp_test.go

```diff
@@ -155,3 +159,40 @@ func (s *snmpDockerSuite) TestSnmp() {
 		assert.Contains(c, metrics, "snmp.sysUpTimeInstance", "metrics %v doesn't contain snmp.sysUpTimeInstance", metrics)
 	}, 5*time.Minute, 10*time.Second)
 }
+
+func (s *snmpDockerSuite) TestSnmpTagsAreStoredOnRestart() {
+	fakeintake := s.Env().FakeIntake.Client()
+	initialMetrics, err := fakeintake.FilterMetrics("snmp.device.reachable",
+		client.WithTags[*aggregator.MetricSeries]([]string{}),
+	)
+	require.NoError(s.T(), err)
+
+	initialTags := initialMetrics[0].Tags
+	_, err = s.Env().RemoteHost.Execute("docker stop dd-snmp")
+	require.NoError(s.T(), err)
+
+	err = fakeintake.FlushServerAndResetAggregators()
+	require.NoError(s.T(), err)
+
+	_, err = s.Env().RemoteHost.Execute(fmt.Sprintf("docker restart %s", s.Env().Agent.ContainerName))
+	require.NoError(s.T(), err)
+
+	err = fakeintake.FlushServerAndResetAggregators()
+	require.NoError(s.T(), err)
+
+	var metrics []*aggregator.MetricSeries
+
+	require.EventuallyWithT(s.T(), func(t *assert.CollectT) {
+		metrics, err = fakeintake.FilterMetrics("snmp.device.reachable",
+			client.WithTags[*aggregator.MetricSeries]([]string{}),
+		)
+		require.NoError(t, err)
+		assert.NotEmpty(t, metrics)
+	}, 5*time.Minute, 5*time.Second)
+
+	tags := metrics[0].Tags
+
+	require.Zero(s.T(), metrics[0].Points[0].Value)
+	require.ElementsMatch(s.T(), tags, initialTags)
+	require.Len(s.T(), tags, 9)
```

Comment:
Consider removing this check since the number of tags could change and it probably doesn't check anything beyond what was already checked by the previous ElementsMatch().

Reply:
removed https://github.com/DataDog/datadog-agent/pull/26003/commits/45ae6cc591eabe4ac00ffe4d6ea11cdb9ff89535